Task.js

```diff
@@ -102,6 +108,51 @@ class Task extends Unit {
     );
   }
 
+  isURL(source) {
+    try {
+      new URL(source);
+      return true;
+    } catch (e) {
+      return false;

```

Comment:
Please log a statement here mentioning that ${source} is not a valid URL. Will help in case an intended URL has errors in it

Reply:
A log statement is added